### PR TITLE
Edit account settings container

### PR DESCRIPTION
**Pull request by Robert-Lark**
# Description

This PR contains the work related to the Edit Account Settings container visible in the screenshot of the prototype below. It will house the edit email and password elements as well as buttons to change subscription status and payment method. 

The container will be locked until the user clicks on a button to unlock the features (this will involve them entering their pin in a pop out modal).

To handle pin validation bcrypt was used as well as react-pin-input. 

figma:
Screen Shot 2021-05-09 at 12 13 40 PM

Coded out UI

Screen Shot 2021-05-10 at 5 32 51 PM
Screen Shot 2021-05-10 at 5 32 39 PM
Screen Shot 2021-05-10 at 5 32 32 PM

Fixes #

- What work was done?
- Container built out and equipped with locking/unlocking functionality. 
- Why was this work done?
- responding to prototype designs from the SS team
- What feature / user story is it for?
- As a user I would like a secure container that houses all the important features of managing my account.
- Any relevant links or images / screenshots

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [ ] Yes
- [x] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: src/components/pages/ParentDashboard/RenderParentDashboard.js [hunk 1]
```diff
@@ -45,4 +45,3 @@ const ParentDashboard = props => {
 };
```
Comment (Robert-Lark) [inline src/components/pages/ParentDashboard/RenderParentDashboard.js:46]: Dont know why this is in my PR - this is from Tuckers nav bar PR

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "react-dom": "^16.13.1",
     "react-image-picker": "^1.1.1",
     "react-pdf": "^5.0.0",
+    "react-pin-input": "^1.0.1",
     "react-plotly.js": "^2.5.0",
```
Comment (Robert-Lark) [inline package.json:32]: react pin input (https://www.npmjs.com/package/react-pin-input) is a small package that is great for handling pin validation - it cut down the amount of code needed to validate - you can compare this pin validation against the other one earlier in the user flow to see the difference.

File: src/__tests__/RenderParentDashboard.test.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ describe('<ParentDashboard /> test suite', () => {
     expect(screen.getByText(/Welcome back/i)).toBeInTheDocument();
   });
 
-  test('Change user Nav', () => {
+  test.skip('Change user Nav', () => {
     render(<Component parent={{ children: [] }} />);
```
Comment (Robert-Lark) [inline src/__tests__/RenderParentDashboard.test.js:45]: After consultation with Sara - these tests (which are testing the old parent dashboard) are skipped to allow a merge.

File: src/components/pages/AccountSettings/RenderAccountSettings.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    setIsModalVisible(false);
+  };
+
+  // this function runs once the user has inputted the correct pin.
+  //It toggles the opacity and disabled prop of the editFormsAndButtonsContainer
+  // allowing the user to see that they can now access the elements to update their account
+
+  const onFinish = values => {
+    console.log('hit');
+    setUnlock(!unlock);
```
Comment (Robert-Lark) [inline src/components/pages/AccountSettings/RenderAccountSettings.js:43]: Whoops - will have to remove that console.log on the next merge

File: src/components/pages/AccountSettings/RenderAccountSettings.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+              if (x == true) {
+                onFinish();
+              } else {
+                setError(true);
+              }
+            }}
+            autoSelect={true}
+          />
+          <p style={error ? null : { display: 'none' }}>Incorrect PIN!</p>
+        </Form>
```
Comment (Robert-Lark) [inline src/components/pages/AccountSettings/RenderAccountSettings.js:84]: Above we have form validation rendered based on an error state - this state is flipped to true if bcrypt cannot validate the pin

**Review by Patrick-G-Oliver — APPROVED** [review]
Looks great.

**Review by StefxM — APPROVED** [review]
loving this!

**Review by cearc-sara — APPROVED** [review]
Looks great

**Pull request merged**